internal/agent/device/osimage.go

```diff
@@ -44,13 +44,12 @@ func (c *OSImageController) Sync(ctx context.Context, desired *v1alpha1.Rendered
 
 	err := c.ensureImage(ctx, desired)
 	if err != nil {
-		if err := c.statusManager.UpdateConditionError(ctx, OsImageDegradedReason, err); err != nil {
-			klog.Errorf("Failed to update condition: %v", err)
+		if updateErr := c.statusManager.UpdateConditionError(ctx, OsImageDegradedReason, err); updateErr != nil {
+			klog.Errorf("Failed to update condition: %v", updateErr)
```

Comment:
I made this change because the way it is, we're returning an error only if we fail setting the condition.